receiver/gitproviderreceiver/internal/scraper/githubscraper/helpers.go

```diff
@@ -206,6 +210,92 @@ func (ghs *githubScraper) getPullRequests(
 	return pullRequests, nil
 }
 
+func (ghs *githubScraper) getCommitInfo(
+	client graphql.Client,
+	repoName string,
+	branch BranchNode,
+) (int, int, int64, error) {
+	comCount := 100
+	var cc *string
+	var adds int
+	var dels int
+	var age int64
+
+	// We're using BehindBy here because we're comparing against the target
+	// branch, which is the default branch. In essence the response is saying
+	// the default branch is behind the queried branch by X commits which is
+	// the number of commits made to the queried branch but not merged into
+	// the default branch. Doing it this way involves less queries because
+	// we don't have to know the queried branch name ahead of time.
+	comPages := getNumPages(float64(100), float64(branch.Compare.BehindBy))
+
+	for nPage := 1; nPage <= comPages; nPage++ {
+		if nPage == comPages {
+			comCount = branch.Compare.BehindBy % 100
+			// When the last page is full
+			if comCount == 0 {
+				comCount = 100
+			}
+		}
+		c, err := ghs.getCommitData(client, repoName, comCount, cc, branch.Name)
+		if err != nil {
+			ghs.logger.Sugar().Errorf("error getting commit data", zap.Error(err))
+			return 0, 0, 0, err
+		}
+
+		if len(c.Edges) == 0 {
+			break
+		}
+		cc = &c.PageInfo.EndCursor
+		if nPage == comPages {
+			e := c.GetEdges()
```

Comment:
This has since changed, however, yes, there's some intricacies on the graphql query -> generated go code side. It's been a while since this code was originally written so I had to go back and re-eval some context. 😅